### PR TITLE
shouldBeEqualToComparingFields handles arrays and computed properties 

### LINKED ISSUE #2375: Improvement of equality.reflection assertions
Hi,

I've played around with some reflect assertions and I think there should be made some changes.

- shouldBeEqualToComparingFields should have an option to skip computed properties (or make skip default and option to add)
- the actually comparing function checkEqualityOfFields could respect some special cases
  - Arrays
  - Others?

If a pull request is welcomed changing these things, I'm willing to provide some code :-)

### PR DESCRIPTION
**Pull request by Kantis**
Should fix #2375 

I didn't quite get the part regarding `bridge` modifier from the issue discussion. What kind of construct is required to create a `bridge` field?

TODO:
- [ ] update changelog?
- [ ] update relevant docs

### CODE REVIEW

File: kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt [hunk 1]
```diff
@@ -383,8 +410,9 @@ private fun <T> checkEqualityOfFields(fields: List<KProperty<*>>, value: T, othe
    return fields.mapNotNull {
       val actual = it.getter.call(value)
       val expected = it.getter.call(other)
-      if (actual == expected) null else {
-         "${it.name}: ${actual.show().value} != ${expected.show().value}"
-      }
+
+      val isEqual = eq(actual, expected) == null
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt:426]: Using `eq` gives us handling of arrays for free.. could it cause other problems? It feels like a nice way to delegate responsibility of equality, IMO

Comment (ashishkujoy) [reply kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/equality/reflection.kt:426]: Using `eq` will be okay, I don't see any problem using it.

**Comment by sksamuel** [comment]
> Should fix #2375
> 
> I didn't quite get the part regarding `bridge` modifier from the issue discussion. What kind of construct is required to create a `bridge` field?

Bridge  and synthetic methods are added by the compiler. So we should just ignore those (if they even show up).

**Comment by sksamuel** [comment]
Don't need to worry about changelog for bugs - I will compile a full list of fixed bugs when I release the next version.
It's good to add details to the changelog / docs for new features tho.

**Pull request merged**